src/handlers/testing/exec/util/comments.ts

```diff
@@ -35,6 +48,227 @@ export async function postSlackMessage(args: {
+        });
+        return;
+      } catch {
+        // This will fail if we don't have permission to comment on PRs,
+        // so we'll continue to commenting on the commit instead
+      }
+    } else {
+      // Create a new comment on the PR
+      try {
+        await api.rest.issues.createComment({
+          owner: github.context.repo.owner,
+          repo: github.context.repo.repo,
+          issue_number: args.ciContext.pullRequestNumber,
+          body: comment,
+        });
+        return;
+      } catch {
+        // This will fail if we don't have permission to comment on PRs,
+        // so we'll continue to commenting on the commit instead
+      }
+    }
+  }
+  const existingCommentId = await findExistingCommentIdOnCommit({
+    githubToken: args.githubToken,
+    commitSha: args.ciContext.commitSha,
+    workflowName: args.ciContext.workflowName,
+    jobName: args.ciContext.jobName,
+  });
+  if (existingCommentId !== null) {
+    // Update the existing comment on the commit
+    await api.rest.repos.updateCommitComment({
+      owner: github.context.repo.owner,
+      repo: github.context.repo.repo,
+      comment_id: existingCommentId,
+      body: comment,
+    });
+    return;
+  } else {
+    // Create a new comment on the commit
+    await api.rest.repos.createCommitComment({
+      owner: github.context.repo.owner,
+      repo: github.context.repo.repo,
+      commit_sha: args.ciContext.commitSha,
+      body: comment,
+    });
+  }
+}
+
+/**
+ * Used to identify comments made by us.
+ *
+ * Assumes users only run one `npx autoblocks testing exec` command per workflow job.
+ */
+function autoblocksGitHubCommentId(args: {
+  workflowName: string;
+  jobName: string;
+}): string {
+  return `<!--- autoblocks-${args.workflowName}-${args.jobName} -->`;
+}
+
+function makeGitHubComment(args: {
+  branchId: string;
+  buildId: string;
+  ciContext: CIContext;
+  runDurationMs: number;
+  evaluations: Evaluation[];
+}): string {
+  const autoblocksUrl = makeAutoblocksCIBuildHtmlUrl({
+    branchId: args.branchId,
+    buildId: args.buildId,
+  });
+  const status = makeTestRunStatusFromEvaluations(args.evaluations);
+  const statusIcon = statusToIcon[status];
+  const statusLabel = statusToLabel[status];
+  const duration = makeDurationString(args.runDurationMs);
+  const buildName = makeBuildName({
+    workflowName: args.ciContext.workflowName,
+    workflowRunNumber: args.ciContext.workflowRunNumber,
+    jobName: args.ciContext.jobName,
+  });
+  const buildUrl = args.ciContext.buildHtmlUrl;
+
+  const statusHeaders = [
+    `${statusIcon} <code>${statusLabel}</code>`,
+    `🕐 ${duration}`,
+    `🏗️ <a href="${buildUrl}">${buildName}</a>`,
+    `➡️ <a href="${autoblocksUrl}">View in Autoblocks</a>`,
+  ].join('&nbsp;&nbsp;•&nbsp;&nbsp;');
+
+  const lines = [
+    `<p align="center"><img src="https://github.com/autoblocksai/cli/assets/7498009/53f0af0b-8ffd-4c95-8dbf-c96051ed5568" width="100%"></p>`,
```

Comment:
not: Maybe pull banner photo into a constant somewhere

Reply:
yeah good call, updated here https://github.com/autoblocksai/cli/pull/66/commits/873458a5f64905d4c8dd504ef9392332e4011bc7